test_cli.py

```diff
@@ -1918,7 +1980,50 @@ def test_workers(ctx):
 
     # Check number of not null values. First 7 columns should be not null. Column "tag" and "runs" could be empty.
     worker_info = lines[2].split()
-    check_equals(True, len(worker_info) >= 9)
+    assert len(worker_info) >= 10
+
+
+@TestModule.register('sharing_workers')
+def test(ctx):
+    def check_workers(user, expected_workers):
+        switch_user(user)
+        result = _run_command([cl, 'workers'])
+
+        # Subtract 2 for the headers that is included in the output of `cl workers`
+        actual_number_of_workers = len(result.split('\n')) - 2
+        check_equals(actual_number_of_workers, len(expected_workers))
+
+        for worker in expected_workers:
+            assert worker in result
+
+    # userA will not start a worker, but will be granted access to one
+    create_user(ctx, 'userA')
+
+    # userB will start a worker and be granted access to another one
+    create_user(ctx, 'userB')
+    create_group(ctx, 'group1')
+    create_worker(ctx, 'userB', 'worker1', group_name='group1')
+
+    # userC will not have access to any workers
+    create_user(ctx, 'userC')
+
+    # userD will start a worker
+    create_user(ctx, 'userD')
+    create_group(ctx, 'group2')
+    add_user_to_group('group2', 'userA')
+    create_worker(ctx, 'userD', 'worker2', group_name='group2')
+
+    # Test to see that a user has access to a worker after the worker was created
+    add_user_to_group('group2', 'userB')
+
+    # Validate user's access to the newly created workers
+    check_workers('userA', ['worker2'])
+    check_workers('userB', ['worker1', 'worker2'])
+    check_workers('userC', [])
+    check_workers('userD', ['worker2'])
+
+    # Switch back to the root user for proper cleanup
+    switch_user('codalab')
```

Comment:
Can we move this line to the `__enter__` method so it's called as a general startup function after all tests?

Reply:
That's cleaner. I updated.